core/src/main/scala/org/http4s/MessageFailure.scala

```diff
@@ -70,6 +70,10 @@ object ParseResult {
   * the entity headers or with the entity itself.   */
 trait DecodeFailure extends MessageFailure
 
+object DecodeFailure {
+  implicit val http4sEqForDecodeFailure: Eq[DecodeFailure] = Eq.fromUniversalEquals
```

Comment:
I don't feel great about this instance.

Reply:
Me neither, but then again I'm not exactly sure whether we care about matching too much whether two decodefailures match, vs simply knowing something failed in a specific way.